sys/shell/shell.c

```diff
@@ -225,7 +225,7 @@ static int readline(shell_t *shell, char *buf, size_t size)
         if (c == '\r' || c == '\n') {
             if (line_buf_ptr == buf) {
                 /* The line is empty. */
-                continue;
+                return 1;
             }
 
             *line_buf_ptr = '\0';
```

Comment:
I don't understand. I expect this:

```
Welcome to the shell!
>
```

Now if I type sth, it should show behind the prompt:

```
Welcome to the shell!
> hello
```

Now if I press enter:
```
Welcome to the shell!
> hello
Hello World!
>
```

Does it behave differently for you?